### PR TITLE
Public Key Hash Extraction From Address

### PR DESCRIPTION
**Pull request by JesseEmond**
Takes an address (e.g. `16UwLL9Risc3QfPqBUvKofHmBQ7wMtjvM`) and decodes the public key hash (HASH160) contained within it using an outside [`Base58Check`](https://en.bitcoin.it/wiki/Base58Check_encoding) library.

### CODE REVIEW

**Comment by DrPandemic** [comment]
> using an outside Base58Check library.

You haven't done it yourself? D:

File: miner/address.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    HASH160 hash (used as a public key hash).
+
+    :see: https://en.bitcoin.it/wiki/List_of_address_prefixes
+
+    :param address: P2PKH public address
+    :returns:       HASH160 hash of the public key
+    """
+    decoded = b58decode_check(address)
+
+    # check that it is a mainnet or testnet P2PKH address
```
Comment (DrPandemic) [inline miner/address.py:16]: Will this break on regtest?

**Comment by DrPandemic** [comment]
If it doesn't break on regtest, :+1:

File: miner/address.py (hunk 1 above)
Comment (JesseEmond) [inline miner/address.py:16]: Not with my manual tests. It's just that [addresses in the mainnet vs testing environments differ in their prefix](https://en.bitcoin.it/wiki/List_of_address_prefixes).

**Comment by JesseEmond** [comment]
Feel free to replace the library with a hand-made implementation (i.e. copying the code from somewhere). I'll gladly deny the PR. :+1: 

;) Merging.

**Pull request merged**